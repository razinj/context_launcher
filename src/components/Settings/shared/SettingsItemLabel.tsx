import React from 'react'
import { StyleProp, StyleSheet, Text, TextStyle, View, ViewStyle } from 'react-native'

export type SettingsItemLabelProps = {
  title: string
  description?: string
  titleStyle?: StyleProp<TextStyle>
  wrapperStyle?: StyleProp<ViewStyle>
}

const SettingsItemLabel = ({ title, description, wrapperStyle, titleStyle }: SettingsItemLabelProps) => {
  return (
    <View style={[styles.wrapper, wrapperStyle]} testID='wrapper'>
      <Text style={[styles.label, titleStyle]} testID='title'>
        {title}
      </Text>
      {description && <Text style={styles.description}>{description}</Text>}
    </View>
  )
}

const styles = StyleSheet.create({
  wrapper: {
    padding: 10,
  },
  label: {
    color: '#808080',
  },
  description: {
    color: '#ccc',
  },
})

export default SettingsItemLabel
