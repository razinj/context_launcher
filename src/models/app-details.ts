export type AppDetails = {
  label: string
  name: string
}

export type AppDetailsWithOptionalIcon = AppDetails & { icon?: string }

export type AppDetailsWithIcon = AppDetails & { icon: string }
