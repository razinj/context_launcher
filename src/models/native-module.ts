export interface AppsModuleInterface {
  getConstants(): AppsModuleConstants
  launchApplication(packageName: string): void
  showApplicationDetails(packageName: string): void
  requestApplicationUninstall(packageName: string): void
  getApplications(callback: (applications: string) => void): void
  getApplicationIcon(packageName: string, callback: (nativeAppIcon: string) => void): void
}

type AppsModuleConstants = {
  appVersion: string
  buildNumber: string
  packageName: string
}
