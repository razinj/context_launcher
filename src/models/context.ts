// React
import { RefObject } from 'react'
// React Native
import { TextInput } from 'react-native'
// BottomSheet
import { BottomSheetModal } from '@gorhom/bottom-sheet'
// Models
import { AppDetailsWithOptionalIcon } from '../models/app-details'

export type GlobalContextType = {
  // All apps
  hideAllApps: () => void
  displayAllApps: boolean
  toggleDisplayAllApps: () => void
  // Settings bottom sheet
  settingsBottomSheetRef: RefObject<BottomSheetModal> | null
  displaySettingsBottomSheet: () => void
  // App item menu
  appItemMenuDetails: AppDetailsWithOptionalIcon | null
  setAppItemMenuDetails: (appDetails: AppDetailsWithOptionalIcon) => void
  appItemMenuBottomSheetRef: RefObject<BottomSheetModal> | null
  displayAppItemMenuBottomSheet: () => void
  // Favorite apps sort
  sortableFavoriteApps: boolean
  toggleSortableFavoriteApps: () => void
  // Global app launch procedure
  globalAppLaunchProcedure: () => void
  // Misc
  dismissKeyboard: () => void
}

export type SearchContextType = {
  // Search input
  searchInputRef: RefObject<TextInput> | null
  searchAppLaunchProcedure: () => void
  // Search query validity
  invalidCharacters: boolean
  setInvalidCharacters: (isInvalid: boolean) => void
}
