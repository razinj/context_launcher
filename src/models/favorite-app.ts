import { AppDetails } from './app-details'

export type FavoriteApp = AppDetails
