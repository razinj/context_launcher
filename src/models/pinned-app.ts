// Models
import { AppDetailsWithIcon } from './app-details'

export type PinnedApp = AppDetailsWithIcon & {
  isPermanent: boolean
  isTemporary: boolean
}

export type TemporaryPinnedAppsConfig = {
  startDate?: string
  endDate?: string
}

export type PinnedAppsState = {
  list: PinnedApp[]
  temporaryAppsConfig: TemporaryPinnedAppsConfig
}
