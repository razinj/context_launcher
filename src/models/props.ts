// React
import { ReactNode } from 'react'
// React Native
import { StyleProp, TextStyle, ViewStyle } from 'react-native'
import { BaseAnimationBuilder } from 'react-native-reanimated'
// Models
import { AppDetails } from './app-details'
import { RenderedIn } from './rendered-in'

export type AppItemProps = {
  appDetails: AppDetails
  renderedIn: RenderedIn
  appIcon?: string
  wrapperStyle?: StyleProp<ViewStyle>
  pressableStyle?: StyleProp<ViewStyle>
}

export type HighlightTextProps = {
  text: string
}

export type SearchContextWrapperProps = {
  children: ReactNode
}

export type GlobalContextWrapperProps = {
  children: ReactNode
}

export type AllAppsLetterIndexProps = {
  onPress: (letterIndex: number) => void
}

export type SettingsItemLabelProps = {
  title: string
  description?: string
  titleStyle?: StyleProp<TextStyle>
  wrapperStyle?: StyleProp<ViewStyle>
}

export type ToggleSettingsProps = SettingsItemLabelProps & {
  children: ReactNode
}

export type CustomViewProps = {
  children: ReactNode
  style?: StyleProp<ViewStyle>
  exitAnimation?: BaseAnimationBuilder | typeof BaseAnimationBuilder
  entryAnimation?: BaseAnimationBuilder | typeof BaseAnimationBuilder
}

export type CustomIconProps = {
  name: string
  size: number
  color: string
  style?: Record<string, string | number>
}
