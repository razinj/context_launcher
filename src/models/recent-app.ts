import { AppDetails } from './app-details'

export type RecentApp = AppDetails // TODO: Remove this type?
