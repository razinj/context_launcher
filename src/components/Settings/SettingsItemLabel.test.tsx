import { renderWithProvider } from '../../../utils/test/utils'
import { SettingsItemLabelProps } from '../../models/props'
import SettingsItemLabel from './SettingsItemLabel'

describe('<SettingsItemLabel /> Tests', () => {
  const requiredProps: SettingsItemLabelProps = {
    title: 'An item label',
  }

  it('should render correctly and match snapshot - required props', () => {
    expect(renderWithProvider(<SettingsItemLabel {...requiredProps} />).toJSON()).toMatchSnapshot()
  })

  it('should render correctly and match snapshot - required and optional props', () => {
    const allProps: SettingsItemLabelProps = {
      ...requiredProps,
      description: 'An item description',
    }

    expect(renderWithProvider(<SettingsItemLabel {...allProps} />).toJSON()).toMatchSnapshot()
  })
})
