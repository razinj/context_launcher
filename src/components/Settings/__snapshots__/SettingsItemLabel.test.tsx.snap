// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsItemLabel /> Tests should render correctly and match snapshot - required and optional props 1`] = `
<View
  style={
    {
      "flexDirection": "column",
      "justifyContent": "center",
      "padding": 10,
    }
  }
>
  <Text
    style={
      {
        "color": "#808080",
      }
    }
  >
    An item label
  </Text>
  <Text
    style={
      {
        "color": "#ccc",
      }
    }
  >
    An item description
  </Text>
</View>
`;

exports[`<SettingsItemLabel /> Tests should render correctly and match snapshot - required props 1`] = `
<View
  style={
    {
      "flexDirection": "column",
      "justifyContent": "center",
      "padding": 10,
    }
  }
>
  <Text
    style={
      {
        "color": "#808080",
      }
    }
  >
    An item label
  </Text>
</View>
`;
